conbench/entities/benchmark_result.py

```diff
@@ -642,36 +643,44 @@ def validate_and_aggregate_samples(stats_usergiven: Any):
     if len(samples) >= 3:
         # See https://github.com/conbench/conbench/issues/802 and
         # https://github.com/conbench/conbench/issues/1118
-        q1, q3 = np.percentile(samples, [25, 75])
+        percentiles: List[float] = list(np.percentile(samples, [25, 75]))
+        q1, q3 = float(percentiles[0]), float(percentiles[1])
 
-        aggregates = {
+        aggregates: Dict[str, float] = {
             "q1": q1,
             "q3": q3,
-            "median": np.median(samples),
+            "median": float(np.median(samples)),
             "min": np.min(samples),
             "max": np.max(samples),
             # With ddof=1 this is Bessel's correction, has N-1 in the divisor.
             # This is the same behavior as
             # statistics.stdev() and the same behavior as
             # scipy.stats.tstd([1.0, 2, 3])
-            "stdev": np.std(samples, ddof=1),
+            "stdev": float(np.std(samples, ddof=1)),
             "iqr": q3 - q1,
         }
 
-        # Now, overwrite with self-derived aggregates:
+        # Now, overwrite with self-derived aggregates.
         for key, value in aggregates.items():
-            result_data_for_db[key] = sigfig.round(value, sigfigs=5)
+            result_data_for_db[key] = value
 
             # Log upon conflict. Let the automatically derived value win, to
             # achieve consistency between the provided samples and the
             # aggregates.
             if key in stats_usergiven:
-                if not floatcomp(stats_usergiven[key], value):
+                try:
+                    if not floatcomp_with_leeway(stats_usergiven[key], value):
+                        log.warning(
+                            "key %s, user-given val %s vs. calculated %s",
+                            key,
+                            stats_usergiven[key],
+                            value,
+                        )
+                except Exception as exc:
                     log.warning(
-                        "key %s, user-given val %s vs. calculated %s",
-                        key,
-                        stats_usergiven[key],
-                        value,
```

Comment:
Also added a bit of safe-guard here, where the `floatcomp_with_leeway()` is just sugar for logging/debugging. It would be sad if a bug in that function was to result in data not being accepted.